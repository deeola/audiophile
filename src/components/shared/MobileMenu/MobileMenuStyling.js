import styled from "styled-components";
import { Colors } from "../../../Themes";

export const MenuContainer = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  margin: 40px auto;
  background: white;
  align-items: center;
  flex-direction: column;
`;

export const SingleMenuContainer = styled.div`
  width: 30%;
  height: 80%;
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  z-index: 2;

  &:before {
    width: 100%;
    height: 70%;
    position: absolute;
    content: "";
    background-color: ${Colors.gray};
    bottom: 0;
    left: 0;
    z-index: -1;
  }

  @media (max-width: 802px) {
    width: ${(props) => (props.mobile ? "30%" : "80%")};
    margin: 20px auto;
  }

  @media (max-width: 490px) {
    width: ${(props) => (props.mobile ? "100%" : "80%")};
  }
`;

export const Text = styled.p``;

export const MenuImage = styled.img`
  width: 50%;
  display: block;
`;

export const MenuCont = styled.div`
  position: absolute;
  z-index: 2;
  top: 30px;
  width: 100%;
  @media (max-width: 802px) {
  }
`;
