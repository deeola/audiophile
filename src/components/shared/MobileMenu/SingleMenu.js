import React from 'react'
import { TetiaryButton } from '../../Buttons/Button';
import { MenuImage, SingleMenuContainer, Text } from './MobileMenuStyling';


const SingleMenu = ({ labels="Shop", label,  alt, span=">", img, link, mobile, datatestid}) => {
  return (
    <SingleMenuContainer data-testid='singlemenu' mobile={mobile}>
        <MenuImage  data-testid='alt' src={img} alt={alt} />
        <Text data-testid='textlabel'>{label}</Text>
        <TetiaryButton datatestid={datatestid} label={labels} link={link} span={span} />
    </SingleMenuContainer>
  )
}

export default SingleMenu;



