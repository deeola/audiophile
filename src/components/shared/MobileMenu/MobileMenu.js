import React from 'react'
import { MenuContainer } from './MobileMenuStyling';
import SingleMenu from './SingleMenu';
import imageheadphone from '../../../assets/shared/desktop/imageheadphones.png';
import imagespeaker from '../../../assets/shared/desktop/imagespeakers.png';
import imageearphone from '../../../assets/shared/desktop/imageearphones.png';

const MobileMenu = ({primary}) => {
  return (
    <MenuContainer data-testid='menu' primary={primary}>
        <SingleMenu  label="HEADPHONES" img={imageheadphone} alt={"headphone"} link={"/heaphones"} />
        <SingleMenu   label="SPEAKERS" img={imagespeaker}  alt={"speaker"} link={"/speakers"} />
        <SingleMenu   label="EARPHONES" img={imageearphone}  alt={"earphone"} link={"/earphones"} />
    </MenuContainer>
  )
}

export default MobileMenu;

