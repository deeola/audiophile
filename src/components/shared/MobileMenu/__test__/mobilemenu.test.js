import React from "react";
import MobileMenu from "../MobileMenu";
import renderer from "react-test-renderer";
import { render, cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";

afterEach(cleanup)

it("renders div correctly", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <MobileMenu />
    </BrowserRouter>,
    div
  );
});

it("renders 3 items", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <MobileMenu data-testid="menu" primary="primary" />
    </BrowserRouter>
  );

  expect(getByTestId("menu").childElementCount).toBe(3);
});


//snapshot

it('matches snapshot', () => {
    const tree = renderer.create(
        <BrowserRouter>
      <MobileMenu data-testid="menu" primary="primary" />
    </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot()
})