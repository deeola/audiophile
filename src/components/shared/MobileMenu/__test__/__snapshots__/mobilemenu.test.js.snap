// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bczRLJ hNDPa-d"
  data-testid="menu"
>
  <div
    className="sc-gsnTZi dnDcju"
    data-testid="singlemenu"
  >
    <img
      alt="headphone"
      className="sc-hKMtZM bESBrL"
      data-testid="alt"
      src="imageheadphones.png"
    />
    <p
      className="sc-dkzDqf"
      data-testid="textlabel"
    >
      HEADPHONES
    </p>
    <a
      className="sc-jSMfEi IOoRh"
      color="tertiary"
      hover="tertiary"
      href="/heaphones"
      onClick={[Function]}
    >
      Shop
      <span>
        &gt;
      </span>
    </a>
  </div>
  <div
    className="sc-gsnTZi dnDcju"
    data-testid="singlemenu"
  >
    <img
      alt="speaker"
      className="sc-hKMtZM bESBrL"
      data-testid="alt"
      src="imagespeakers.png"
    />
    <p
      className="sc-dkzDqf"
      data-testid="textlabel"
    >
      SPEAKERS
    </p>
    <a
      className="sc-jSMfEi IOoRh"
      color="tertiary"
      hover="tertiary"
      href="/speakers"
      onClick={[Function]}
    >
      Shop
      <span>
        &gt;
      </span>
    </a>
  </div>
  <div
    className="sc-gsnTZi dnDcju"
    data-testid="singlemenu"
  >
    <img
      alt="earphone"
      className="sc-hKMtZM bESBrL"
      data-testid="alt"
      src="imageearphones.png"
    />
    <p
      className="sc-dkzDqf"
      data-testid="textlabel"
    >
      EARPHONES
    </p>
    <a
      className="sc-jSMfEi IOoRh"
      color="tertiary"
      hover="tertiary"
      href="/earphones"
      onClick={[Function]}
    >
      Shop
      <span>
        &gt;
      </span>
    </a>
  </div>
</div>
`;
